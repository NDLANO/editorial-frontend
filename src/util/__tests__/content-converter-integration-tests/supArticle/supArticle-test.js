/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import jsdom from 'jsdom';
import prettify from '../prettify';
import {
  learningResourceContentToEditorValue,
  learningResourceContentToHTML,
} from '../../../articleContentConverter';
import { html } from './supArticle';

const { fragment } = jsdom.JSDOM;

test('serializing article with sup tag', () => {
  const converted = learningResourceContentToEditorValue(html, fragment);

  const result = learningResourceContentToHTML(converted);

  expect(prettify(result)).toMatchSnapshot();
});
