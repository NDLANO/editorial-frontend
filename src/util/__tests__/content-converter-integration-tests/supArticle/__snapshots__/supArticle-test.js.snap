// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializing article with sup tag 1`] = `
"<section>
  <h2>Opprinnelig friksjonstrykk i systemet</h2>
  <p>
    Brønnen er 2500 m dyp, og casingskoen er satt ved 2300 m. <br />Boreslammets
    densitet er 1,20 sg. Friksjonstrykkgradienten er 0,02 bar/meter i borerørene
    og 0,001 bar/meter i ringrommet ved 40 spm. Trykkfallet gjennom dysene er 20
    bar.
  </p>
  <p>1) Det hydrostatiske trykket i bunnen av brønnen er:</p>
  <p><strong>phydBH = 1,20 ∙ 0,0981 · 2500 = 294,3 bar</strong></p>
  <p>
    2) Når pumpen leverer 40 spm, er trykket på overflaten det sammenlagte
    trykket i systemet:
  </p>
  <p>ppumpe1 = pfdp + pfdyser + pfr</p>
  <p>pfdp = 2500 · 0,02 = 50 bar</p>
  <p>pfdyser = 20 bar</p>
  <p>pfr = 2500 · 0,001 = 2,5 bar</p>
  <p><strong>Sum friksjonstrykk: 50 + 20 + 2,5 = 72,5 bar</strong></p>
  <p>3) Sirkulasjonstrykket i bunnen av brønnen:</p>
  <p>pBH1 = phyd + pfr</p>
  <p>
    <strong>pBH1 = (2500 · 0,0981 · 1,20) + 2,5 = 296,8 bar</strong
    ><strong><br /></strong>
  </p>
  <h2>Økning i væskestrøm</h2>
  <p>Vi fortsetter med å øke pumperaten til 80 SPM.</p>
  <p>4) Sirkulasjonstrykket på overflaten viser nå:</p>
  <p>pf2 = pf1 (Q2/Q1)<sup>2</sup></p>
  <p>
    <strong>pf2 = 72,5 · (80/40)</strong><strong><sup>2</sup></strong
    ><strong> = 290 bar</strong>
  </p>
  <p>5) Sirkulasjonstrykket i bunnen av brønnen:</p>
  <p>pBH2 = pfr2 + phyd1</p>
  <p>
    <strong>pBH2</strong> = <strong>2,5 · (80/40)</strong
    ><strong><sup>2</sup></strong
    ><strong> + </strong><strong>294,3 = 304,3 bar</strong
    ><strong><br /></strong>
  </p>
</section>
<section>
  <h3><em>Fordypning</em></h3>
  <p>
    Nå er endringen på grunn av høyere pumperate. Her bruker vi formelen pf2 =
    pf1 (Q2/Q1)<sup>2</sup>
  </p>
  <div class=\\"c-bodybox\\">
    <p>i) Ny friksjonsfaktor i ringrommet:</p>
    <p>pfr2 = 2,5 · (80/40)<sup>2 </sup>= 10 bar</p>
    <p>
      <strong>Friksjonsgradient: 10/2500 = 0,004 bar/meter</strong
      ><strong><br /></strong>
    </p>
    <p>ii) Ny friksjonsfaktor i borestrengen:</p>
    <p>pfdp2 = 50 bar · (80/40)<sup>2</sup> = 200 bar</p>
    <p>
      <strong>Friksjonsgradient: 200/2500 = 0,008 bar/meter</strong
      ><strong><br /></strong>
    </p>
    <p>iii) Ny friksjon i dysene:</p>
    <p>
      <strong>pfdyser2 = 20 · (80/40)</strong><strong><sup>2</sup></strong
      ><strong> = 80 bar </strong><strong><br /></strong>
    </p>
  </div>
</section>
<section>
  <h2>Endring i boreslammets densitet</h2>
  <p>Boreslammets densitet er 1,25 sg, og pumperaten er 80 SPM.</p>
  <p>6) Trykket på overflaten:</p>
  <p>pf2 = pf1 · (d2/d1)</p>
  <p><strong>pf2 = 290 · (1,25/1,20) = 302 bar</strong></p>
  <p>7) Sirkulasjonstrykket i bunnen av brønnen:</p>
  <p>pBH3 = phyd2 + pfr3</p>
  <p>
    <strong>pBH3 = (2500 · 0,0981 · 1,25) + 10 · (1,25/1,20) = 317 bar</strong>
  </p>
</section>
<section>
  <h3><em>Fordypning</em></h3>
  <p>
    Her regner vi ut hydrostatisk trykk til casingskoen og friksjonstrykket i
    ringrommet over casingskoen.
  </p>
  <div class=\\"c-bodybox\\">
    <p>phydcs = 2300 · 0,0981 · 1,25 = 282 bar</p>
    <p>
      pfrcs = pfr3 · (2300/2500) = 10 · (1,25/1,20) · (2300/2500) = 9,59 bar
    </p>
    <p><strong>Samlet trykk ved casingskoen: 282 + 9,59 = 291,6 bar</strong></p>
  </div>
</section>
<section>
  <h2>Brønnens lengde øker</h2>
  <p>
    Boringen går framover, og brønnen er <strong>3200 meter </strong>lang.
    Boreslammets densitet er <strong>1,25 sg, </strong>og vi sirkulerer med
    sakte rate for å måle friksjonen i systemet, <strong>40 SPM</strong>.
  </p>
  <p>
    Vi regner ut friksjonen på 3200 meter med friksjonsfaktoren for 40 spm med
    1,20 sg og multipliserer med forholdet for økning av densitet.
  </p>
  <p>8) Nytt friksjonstrykk:</p>
  <p>ppumpe = pfdp3 + pfdyser3 + pfr4</p>
  <p>pfdp3 = 3200 · 0,02 · (1,25/1,20) = 67 bar</p>
  <p>pfdyser3 = 20 · 1,25/1,20 = 21 bar</p>
  <p>pfr4 = 3200 · 0,001 · (1,25/1,20) = 3,3 bar</p>
  <p><strong>Samlet friksjonstrykk i systemet: 91 bar</strong></p>
  <p>9) Trykket i bunnen av brønnen:</p>
  <p>pBH4 = phyd3 + pfr4</p>
  <p>phyd3 = 3200 · 0,0981 · 1,25 = 392,4 bar</p>
  <p>pfr4 = 3,3 bar</p>
  <p><strong>Samlet bunntrykk: 396 bar</strong></p>
</section>
<section>
  <h3><em>Fordypning</em></h3>
  <p>
    Samlet bunnhullstrykk kan uttrykkes i densitet og heter ECD når vi
    sirkulerer:
  </p>
  <div class=\\"c-bodybox\\">
    <p>PBH5 = 392,4 + 3,3 · (80/40)<sup>2</sup> = 405,6 bar</p>
    <p>p = d · g · h</p>
    <p>d = p/(g · h)</p>
    <p><strong>dECD = 405,6/(0,0981 · 3200) = 1,292 sg ≈ 1,30 sg</strong></p>
  </div>
</section>
<section>
  <div data-type=\\"related-content\\">
    <embed data-size=\\"2\\" data-__altered=\\"false\\" /><embed
      data-size=\\"2\\"
      data-__altered=\\"false\\"
    />
  </div>
</section>
"
`;
